### DISCUSSION TITLE
Cyberduck Can't Find Local File?

### DISCUSSION THREAD

**Discussion by WhatImListnTo** [iterate-ch/cyberduck#16076] [Q&A]
I connected to an SFTP server, navigated to an empty directory on the server then clicked the Upload button.  I have over 30,000 directories and files combined that exist on my local D: drive:  D:\Media\Digital Audio.  I selected all content within that directory to then upload to the SFTP server.  Cyberduck started creating all directories.  During this directory creation and before any file uploads, I get a strange error:

![cyberduck_error](https://github.com/iterate-ch/cyberduck/assets/55523642/38bfb9af-205e-4c0f-9679-f8e728735618)

This file it is claiming that it cannot find is on my local D: drive and is valid.  Why would Cyberduck be complaining about a local file referring me to my web hosting service provider?  This doesn't make sense.

**Comment by WhatImListnTo** [comment]
I have the latest FLAC.EXE 64-bit command-line installed on my Windows 11 computer.  I just used the test command line parm against the file in question on my D: drive:

![flacui_validate](https://github.com/iterate-ch/cyberduck/assets/55523642/55553551-355b-4a9a-bf62-a5d9e3b60591)

Below are the results:

![flac_validation_results](https://github.com/iterate-ch/cyberduck/assets/55523642/6536692a-2878-483d-afa3-d41cb70d39a4)

This is Cyberduck 64-bit version 8.9.0 (41543).

**Comment by WhatImListnTo** [comment]
Incidentally, the reason that I am performing this upload with Cyberduck is that my Mountain Duck is foobar and will not complete correctly.  My plan is to perform the entire 1.2 TiB upload using Cyberduck, completely uninstall both Cyberduck and Mountain Duck deleting all cached files, and lastly re-install both again for a fresh start.  I will then use Mountain Duck to sync the complete upload done by Cyberduck.

My plan is to have a locally cached sync to play content from my Windows 11 computer.

**Comment by tatze96** [comment]
Regarding the "File not Found" error:
Can you please [enable debug logging](https://docs.cyberduck.io/cyberduck/support/#debug-logging-level), recreate the error message, and send us the log file afterward? If you don't like to see sensitive data to be shared publicly, please send the log file with the ticket reference GH-16076 to support@cyberduck.io, otherwise attach it here.

**Sidenote**: If you want to share the log file here, make sure to attach it through the GitHub web interface, as it wouldn't be visible to us otherwise.

Regarding your use case:
If you use the [offline functionality](https://docs.cyberduck.io/mountainduck/connect/sync/#keep-offline) of Mountain Duck, you'll still need the mounted drive to access the files as the synchronization cache is obfuscated.

**Comment by WhatImListnTo** [comment]
Thank you for emailing.

The issue I faced with CyberDuck upload was on an SFTP server from a 
service provider a discontinued service with.  I was able to reproduce 
the issue quickly, but after canceling service I would not be able to.  
The service provider was IONOS.

I am getting a new SFTP server that I am working to order now.  It 
should be available shortly, and I will attempt to reproduce the issue 
on the new SFTP server once it is ready.

Regarding the Mountain Duck offline functionality note you added, my 
plan was to completely upload all data via Cyberduck to the SFTP server. 
  Once that is done, I will use Mountain Duck to then download everything 
for offline mode to be kept in sync.  I would not refer to the 
obfuscated data later and will have no need to.  As long as the sync 
works properly, I will work directly with the mapped drive only.  This 
should work...unless it doesn't later.  I would post about future issues 
if faced later.


------ Original Message ------
From "Laura" ***@***.***>
To "iterate-ch/cyberduck" ***@***.***>
Cc "WhatImListnTo" ***@***.***>; "Author" 
***@***.***>
Date 7/5/2024 7:30:03 AM
Subject Re: [iterate-ch/cyberduck] Cyberduck Can't Find Local File? 
(Discussion #16076)

**Comment by WhatImListnTo** [comment]
I got my new SFTP server.  I attempted what failed previously again, and I got a worse failure this time uploading to new server.  I selected the option to send a crash log.  I am not sure what went wrong.  However, I successfully uploaded 16 files.  After this, I successfully selected and uploaded a single selected directory containing a nested directory and several files underneath.  After this completes, I will again select a large number of directories with all files under to see what happens

**Comment by WhatImListnTo** [comment]
It looks like the issue is that I am uploading too much at once.  The transfer gets canceled with such a large number, and Cyberduck does not recover.  I have to close the application, restart it, then select a smaller number of directories.  It did not get through the directory creation portion and never started uploading files.

**Comment by WhatImListnTo** [comment]
My logs have been overwritten it appears, and I lost the debug log of the crash I stated earlier.  My upload is continuing, and Cyberduck has no issues uploading 229.3GiB content at a time.  I will continue increasing the  upload size to determine what causes the failure.  My current upload will be complete tomorrow.

I do have the crash log output, but I am not sure how helpful this will be:

```
-----------------------------
[General Info]

Application: Cyberduck
Version:     9.0.0.41777+10a5fb9d32e92c0422eccaf7936efce7d9cb4dd0
Region:      English (United States)
Date: 7/5/2024
Time: 11:45 PM

User Explanation:

User said ""
-----------------------------

[Exception Info 1]

Top-level Exception
Type:        java.util.ConcurrentModificationException
Message:     
Source:      IKVM.OpenJDK.Core
Stack Trace: at java.util.ArrayList.Itr.checkForComodification()
   at java.util.ArrayList.Itr.next()
   at ikvm.lang.IterableEnumerator.MoveNext()
   at Ch.Cyberduck.Ui.Controller.TransferController.ApplicationShouldTerminate()
   at Ch.Cyberduck.Ui.Controller.MainController.ApplicationShouldTerminate()
   at Ch.Cyberduck.Ui.Controller.MainController.<>c.<NewBrowser>b__55_0(Object sender, FormClosingEventArgs args)
   at System.Windows.Forms.FormClosingEventHandler.Invoke(Object sender, FormClosingEventArgs e)
   at Ch.Cyberduck.Ui.Winforms.BaseForm.<.ctor>b__3_5(Object sender, FormClosingEventArgs args)
   at System.Windows.Forms.FormClosingEventHandler.Invoke(Object sender, FormClosingEventArgs e)
   at System.Windows.Forms.Form.OnFormClosing(FormClosingEventArgs e)
   at System.Windows.Forms.Form.WmClose(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

-----------------------------

[Assembly Info]

mscorlib, Version=4.0.0.0
PresentationFramework, Version=4.0.0.0
PresentationCore, Version=4.0.0.0
System, Version=4.0.0.0
System.Windows.Forms, Version=4.0.0.0
System.Drawing, Version=4.0.0.0
Cyberduck.Core.Refresh, Version=9.0.0.41777
Cyberduck.Core.Native, Version=9.0.0.41777
System.Memory, Version=4.0.1.2
System.Xaml, Version=4.0.0.0
WindowsBase, Version=4.0.0.0
CommunityToolkit.Mvvm, Version=8.2.0.0
System.Core, Version=4.0.0.0
Cyberduck.Core, Version=9.0.0.41777
DynamicData, Version=7.5.0.0
System.Reactive, Version=5.0.0.0
IKVM.OpenJDK.Core, Version=8.0.312.7
System.ServiceModel, Version=4.0.0.0
StructureMap, Version=2.6.4.0
Splat, Version=14.1.0.0
ObjectListView, Version=2.9.1.1072
CustomOpenFileFolderDialog, Version=0.0.0.0
WindowsFormsIntegration, Version=4.0.0.0
System.Design, Version=4.0.0.0
System.Runtime.CompilerServices.Unsafe, Version=6.0.0.0
Cyberduck.Bonjour.Native, Version=9.0.0.41777
System.Security, Version=4.0.0.0
Cyberduck.Protocols, Version=9.0.0.41777
ReactiveUI, Version=17.1.0.0
Cyberduck.Importer, Version=9.0.0.41777
System.Xml, Version=4.0.0.0
Cyberduck.Bonjour, Version=9.0.0.41777
IKVM.Runtime, Version=8.0.312.7
ReactiveUI.Wpf, Version=17.1.0.0
Cyberduck.Cryptomator, Version=9.0.0.41777
IKVM.OpenJDK.Security, Version=8.0.312.7
IKVM.OpenJDK.Text, Version=8.0.312.7
Windows.Services.Store.StoreContract, Version=4.0.0.0
System.Web, Version=4.0.0.0

-----------------------------

[System Info]

Operating System
-Microsoft Windows 11 Pro
--CodeSet = 1252
--CSDVersion = 
--CurrentTimeZone = -240
--FreePhysicalMemory = 177799028
--OSArchitecture = 64-bit
--OSLanguage = 1033
--ServicePackMajorVersion = 0
--ServicePackMinorVersion = 0
--Version = 10.0.22631

Machine
-DAEMON
--Manufacturer = ASUS
--Model = System Product Name
--TotalPhysicalMemory = 205886771200

-----------------------------
```

**Comment by WhatImListnTo** [comment]
It looks like once you set debug that logs are only kept for five ZIP backups.  After five, the new logs replace the older ZIPs after five.

**Comment by WhatImListnTo** [comment]
The issue I am facing appears to be specific to Cyberduck attempting to create all directories prior to upload in both this latest case and the previous one.  It seems to be doing this too fast that results in failure.

I am wondering if instead you were to create directories if they do not exist just prior to upload as opposed to blast creating all at the beginning.  For example, 50 nested directories exist in total with various files all within this directory hierarchy.  I select all directories under a parent.  Like this:

directory_1
  +sub_drectory_a
    -file_1
    -file_2
  +sub_drectory_b
    -file_3

This is a small example above.  What currently happens is that Cyberduck is mass creating directories:  directory_1, sub_drectory_a, sub_drectory_b.  Only after this is done are files ever then uploaded.  Why not instead work on file_1, and if directory_1 does not exist, create it.  If sub_drectory_a does not exist under directory_1, create it.  With both directories existing, only then upload file_1.  Repeat this same logic for file_2 in which case no directory creation would be necessary.

**Comment by WhatImListnTo** [comment]
My SFTP upload of  229.3GiB just completed successfully.  Next I attempted my next largest SFTP upload of 723GiB.  This one failed right away after a a new logon:

![07-07-2024_SFTP_FAIL](https://github.com/iterate-ch/cyberduck/assets/55523642/2a32848f-e66d-4ff0-b19e-eaa17f0a9cc7)

I will email the captured debug log for investigation.

Not a single directory was created on the destination after my previous successful SFTP upload pictured above.

**Comment by WhatImListnTo** [comment]
My next several attempts also failed.  I am finally getting a successful upload of 77.1GiB right now.  This should complete in less than ten hours.

**Comment by WhatImListnTo** [comment]
Also, I had to close Cyberduck followed by re-launching it to start the upload.  The job was stuck and would not stop.

**Comment by WhatImListnTo** [comment]
I think the job is failing due to Cyberduck spamming directory creation.

**Comment by WhatImListnTo** [comment]
I just reviewed jorunalctl in its entirety.  I see nothing bad related to this activity in the logs of my server.

**Comment by WhatImListnTo** [comment]
I added an option of firewall to my server, so it could be Cyberduck is being mistaken for a DDoS attack.

**Comment by WhatImListnTo** [comment]
Surprisingly, my smaller upload just failed:
![07-07-2024_SFTP_FAIL2](https://github.com/iterate-ch/cyberduck/assets/55523642/4d6fd5b6-0f1c-4019-af3c-98f37b84e313)

I successfully resumed the upload via clicking Resume.

Debug log emailed under same ticket number.

**Comment by WhatImListnTo** [comment]
Here is the UI after resume:
![07-07-2024_SFTP_RESUMED](https://github.com/iterate-ch/cyberduck/assets/55523642/853b0e4d-cbf1-4708-8057-6dc91703db6b)

I will perform a flac test on the file after it is uploaded to ensure the resume worked correctly.

**Comment by WhatImListnTo** [comment]
This 77.1GiB failure may have been my mistake.  I accidentally set the upload path to /sftp instead of /sftp/Music.  I noticed this yesterday and started moving complete directories already uploaded from /sftp to /sftp/Music.  I moved the entire sub-directory complete with all files and nested sub-directories.  I originally thought Cyberduck was working in alphabetic order, but it appears some folders I moved were empty and not yet uploaded.

I canceled this upload and started a new upload job to the correct directory overwriting any existing files.  It completed the directory creation and is now uploading files.  This upload is 262.28GiB:

![07-07-2024_SFTP_START](https://github.com/iterate-ch/cyberduck/assets/55523642/99abf7b4-03f0-4db6-ab4a-fcbf5c53350b)

This upload should finish either tomorrow or the day after.

**Comment by WhatImListnTo** [comment]
I see why it is starting at the wrong directory and will correct my bookmark to prevent this from happening again.

**Comment by WhatImListnTo** [comment]
My above upload completed, and I hit another immediate failure.  The app is stuck now.  No buttons will work:
![07-08-2024_SFTP_FAIL](https://github.com/iterate-ch/cyberduck/assets/55523642/169ec3db-c41c-4bcd-b959-5a39aec8d84c)

I am emailing the latest log.

**Reply by WhatImListnTo** [reply]
I can no longer use Cyberduck for my SFTP upload.  It will no longer upload any files regardless of how small now.  I must use other SFTP clients to complete my upload.  Other SFTP clients are working without issue.  Additionally, I contacted support for my server and they confirmed there are no limits on their end.  It is a Cyberduck issue specifically.

**Reply by tatze96** [reply]
Please set the _Transfer Files_ dropdown within the Cyberduck preferences _Transfers_ tab to `Open single connection` instead of `Open multiple connections` to see if this improves the upload behavior.

**Comment by WhatImListnTo** [comment]
I attempted this yesterday as well as many other things.  This did not 
help.  The uploads continued to fail immediately.

This is a peculiar issue.  It started off working with 157 total selections on the first successful upload.  I repeated 157 on the next attempt.  On the third attempt I got instant failures with no uploads.  I had to reduce the selection count from 156 down to 36 to have a successful upload.  After 36 completed, I was never able to upload again using Cyberduck.

------ Original Message ------
From "Laura" ***@***.***>
To "iterate-ch/cyberduck" ***@***.***>
Cc "WhatImListnTo" ***@***.***>; "Author" 
***@***.***>
Date 7/9/2024 6:32:26 AM
Subject Re: [iterate-ch/cyberduck] Cyberduck Can't Find Local File? 
(Discussion #16076)

**Comment by WhatImListnTo** [comment]
Also, the failures are instant now.  There is a delay of about one or two seconds then I immediately see failure.  I am over 60% my current upload (I still need to get this done).  I can try again probably tomorrow using Cyberduck with more logging.

**Comment by WhatImListnTo** [comment]
I did two things -- I ran Windows Update, and I also rebooted my computer.  I did not set the preference to use a single connection.  However, I do currently have the Connections drop down on the Transfer screen to 1 Connection.

I selected the first missing folder from local followed by selecting the absolute last folder.  It is uploading now, and when this completes it will be done:

![07-09-2024_SFTP_FINAL_UPLOAD](https://github.com/iterate-ch/cyberduck/assets/55523642/7ab733b4-881a-4c77-a3a3-1b9a064d7019)

Note that I normally do not reboot or power off my computer.  It stays on constantly.  I wonder if the reboot may have resolved this issue?  I do recall seeing posts about memory leaks.

**Comment by WhatImListnTo** [comment]
Bad news -- it just failed.  I do not know why it would have failed this time.  Everything should have worked perfectly.

![07-10-2024_SFTP_FAIL](https://github.com/iterate-ch/cyberduck/assets/55523642/6cbf4143-91b4-41cf-b1ce-bb824c32e7a6)

I will email the latest log.

**Comment by WhatImListnTo** [comment]
I clicked the resume button, and it is now continuing.

It appears that the default timeout and retry settings are too low.  I increased both of these.

Is it also possible to auto-resume a failure after a delay?

**Comment by WhatImListnTo** [comment]
This failed yet again.  My expectation was that this completed today, but it failed:

![07-10-2024_SFTP_FAIL2](https://github.com/iterate-ch/cyberduck/assets/55523642/9f85c34a-9c4a-4360-89f1-d32e92ac1214)

I will email captured log again.  Also, I will attempt to resume again.

**Comment by WhatImListnTo** [comment]
Unfortunately, debug logs spams so much that I have no way of knowing which ZIP file to send.

**Comment by WhatImListnTo** [comment]
I have turned off debug logging considering it has not been useful for any issues I am facing.

**Comment by WhatImListnTo** [comment]
Cyberduck for a small number of file transfers works fine without issue.  It is not ideal for large (over 1 TiB) transfers at this point in time.  Lots of time gets spent clicking resume over and over, and you will have to reboot your computer when the transfers eventually all fail.

**Comment by WhatImListnTo** [comment]
I am finally almost finished my upload:
![07-12-2024_SFTP_ALMOST_DONE](https://github.com/user-attachments/assets/c79f394d-8f7e-4ff6-af1c-48c8520ebb3e)

The progress shows 100%, but it is still uploading.  It appears several files are still missing from the server.  The total space calculation looks off.

Also, I noticed duplicate files were created on the server.  I am not sure how this happened or how this was even possible.  There were about three sets of the same file existing as duplicates in three different directories.  I deleted all of these files followed by uploading the single file in all three directories.  This may have been caused by resuming, but I am not sure how this was even possible.

**Comment by WhatImListnTo** [comment]
It looks like the upload finally completed.  My shell tree list shows zero files missing.  Unfortunately, Cyberduck will not stop uploading.  It appears to be re-uploading files that previously completed.  I just forced stopped it when re-uploading a completed file.  This single file is now corrupt.  I will delete then upload this single one and finally be done with this.

This 1.15TiB upload took seven days.  It would have been shorter if not for all the issues I faced.
